src/Tribe/Rewrite.php

```diff
@@ -676,4 +693,42 @@ protected function get_option_controlled_slug_entry( array $localized_matchers,
 
 		return $entry;
 	}
+
+	/**
+	 * Returns the map of localized bases for the specified text domains.
+	 *
+	 * The bases are the ones used to build the permalinks, the domains are those of the currently activated plugins
+	 * that include a localized rewrite component.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string> $bases   The bases to set up the locale translation for.
+	 * @param array<string> $domains A list of text domains belonging to the plugins currently active that handle and
+	 *                               provide support for a localized rewrite component.
+	 *
+	 * @return array<string,string> A map relating the bases in their English, lowercase form to their current locale
+	 *                              translated form.
+	 */
+	public function get_localized_bases( array $bases, array $domains ) {
+		$locale             = get_locale();
+		$cache_key          = __METHOD__ . md5( serialize( array_merge( $bases, $domains, [ $locale ] ) ) );
+		$expiration_trigger = Cache_Listener::TRIGGER_GENERATE_REWRITE_RULES;
+
+		$cached = tribe_cache()->get( $cache_key, $expiration_trigger, false );
+
+		if ( false !== $cached ) {
+			return $cached;
+		}
+
+		$localized_bases = tribe( 'tec.i18n' )->get_i18n_strings_for_domains( $bases, [ $locale ], $domains );
+
+		$return = array_filter( array_map( static function ( $locale_base ) {
+			return is_array( $locale_base ) ? end( $locale_base ) : false;
+		}, $localized_bases
+		) );
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 3b4e862399ea8d1027d9fa60190411c4</sub>